File: DevExpress-Demo/CS/PropertyGridDemo.Wpf/Data/AttributesSupportData.cs
using DevExpress.Mvvm.DataAnnotations;
using System;
using System.ComponentModel;

namespace PropertyGridDemo {
    public class AttributesSupportData {
        [Browsable(false)]
        public int ID { get; set; }

        [DefaultValueAttribute(18)]
        [NumericMask(Mask = PredefinedMasks.Numeric.Decimal + "2")]
        public int Age { get; set; }

        [DateTimeMask(Mask = PredefinedMasks.DateTime.LongDate)]
        public DateTime HireDate { get; set; }

        const string SSNMask = @"\d{3}-\d{2}-\d{4}";
        [RegExMask(Mask = SSNMask)]
        public string SSN { get; set; }

        [DisplayName("First name"), RefreshProperties(RefreshProperties.All)]
        public string FirstName { get; set; }

        [DisplayName("Last name"), RefreshProperties(RefreshProperties.All)]
        public string LastName { get; set; }

        [DisplayName("Full name")]
        public string FullName {
            get {
                string delimiter = string.IsNullOrEmpty(FirstName) ? string.Empty : " ";
                return FirstName + delimiter + LastName;
            }
        }

        [ReadOnlyAttribute(true)]
        public Gender? Gender { get; set; }
    }
}

File: DevExpress-Demo/VB/PropertyGridDemo.Wpf/Data/AttributesSupportData.vb
Imports DevExpress.Mvvm.DataAnnotations
Imports System.ComponentModel

Namespace PropertyGridDemo

    Public Class AttributesSupportData

        <Browsable(False)>
        Public Property ID As Integer

        <DefaultValueAttribute(18)>
        <NumericMask(Mask:=PredefinedMasks.Numeric.Decimal & "2")>
        Public Property Age As Integer

        <DateTimeMask(Mask:=PredefinedMasks.DateTime.LongDate)>
        Public Property HireDate As Date

        Const SSNMask As String = "\d{3}-\d{2}-\d{4}"

        <RegExMask(Mask:=SSNMask)>
        Public Property SSN As String

        <DisplayName("First name"), RefreshProperties(RefreshProperties.All)>
        Public Property FirstName As String

        <DisplayName("Last name"), RefreshProperties(RefreshProperties.All)>
        Public Property LastName As String

        <DisplayName("Full name")>
        Public ReadOnly Property FullName As String
            Get
                Dim delimiter As String = If(String.IsNullOrEmpty(FirstName), String.Empty, " ")
                Return FirstName & delimiter & LastName
            End Get
        End Property

        <ReadOnlyAttribute(True)>
        Public Property Gender As Gender?
    End Class
End Namespace
